pkg/kine/drivers/generic/admission.go

```diff
@@ -0,0 +1,60 @@
+package generic
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"golang.org/x/sync/semaphore"
+)
+
+// AdmissionControlPolicy interface defines the admission policy contract.
+type AdmissionControlPolicy interface {
+	// Admit checks whether the query should be admitted.
+	// If the query is not admitted, a non-nil error is returned with the reason why the query was denied.
+	// If the query is admitted, then the error will be nil and a callback function is returned to the caller.
+	// The caller must execute it after finishing the query
+	Admit(context.Context) (callOnFinish func(), err error)
+}
+
+// AllowAllPolicy always admits queries.
+type AllowAllPolicy struct{}
+
+// Admit always admits requests for AllowAllPolicy.
+func (p *AllowAllPolicy) Admit(context.Context) (func(), error) {
+	return func() {}, nil
+}
+
+// AdmissionMetrics stores admission-related metrics.
+// TODO(MK-1398): Integrate those metrics with Prometheus.
+type LimitPolicyMetrics struct {
+	// NumCurrentTxn stores the number of currently running transactions.
+	NumCurrentTxn atomic.Int64
```

Comment:
any reason why this should be public? let's maybe fully remove the unused metrics atm, and work on adding a prometheus counter in a next PR

Reply:
Good point - removed